src/components/TextClassification/index.js

```diff
@@ -23,6 +24,20 @@ export const TextClassification = (props) => {
       typeof l === "string" ? { id: l, description: l, displayName: l } : l
     )
     .map((l) => (!l.displayName ? { ...l, displayName: l.id } : l))
+
+  const [textToShow, changeTextToShow] = useState("")
+  const oldText = useRef()
```

Comment:
try to avoid using refs, they should be used as a last resort. I believe you can do this with `useState`.

Also this code will break if `TextClassification` has multiple samples (the next sample textUrl won't be loaded)

Reply:
I would actually track what sampleIndex was loaded and prevent the useEffect from firing if the sampleIndex loaded is the same.

Also: You should early-return out of functions. Don't surround functions in an `if` statement (it keeps nesting down and makes code more readable!)

```javascript
const [loadedSampleIndex, setLoadedSampleIndex] = useState(null)
useEffect(() => {
    if (loadedSampleIndex === currentSampleIndex) return
    setLoadedSampleIndex(currentSampleIndex)
      getTextfromSample(props.samples[currentSampleIndex])
        .then((result) =>  changeTextToShow(result))
  }, [props.samples, currentSampleIndex, loadedSampleIndex])
```

Reply:
Ok i will do the changes in the second comment but for the first. I know another possibility but it slows the app astonishly, so i'm hesitant to turn back to the old way. (the old way = Load all the text and set document in the json on the moment we get the url)

Reply:
In fact i prevent it to load again in getTextfromSample

Reply:
In fact would it be better to just set document ?

Reply:
Is the `textUrl` being written to the `document` of the sample somewhere?

Reply:
I'm not sure I understand what is slowing the app down in your above comment. We shouldn't load all the samples at once, however we will speed it up with predictive caching later (where we load urls we expect we'll need to load)

Reply:
"samples": [
        {
          "document": "This strainer makes a great hat, I'll wear it while I serve spaghetti!",
          "textUrl": "https://fakeUrl"
        },
        { "document": "Why are all these dumpings covered in butter?!" }
      ]

Reply:
It can be a nice idea but the API in get-text-from-url(new version) or dataset-helper.getTextFromUrl is really time consumming. So it's better to load them one by one.

Reply:
On what type of data will we base our predictive casting (the file used ?)